lib/index.js

```diff
@@ -474,21 +474,25 @@ class AuthenticationPlugin {
   /**
    * Search users by ES query concerning exclusively their username
    *
-   * @param {Object} query
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (undefined), size (undefined)
    *
-   * @returns {Promise.<{ scrollId, hits, aggregations, total }>}
+   * @returns {Promise.<{ hits, aggregations, suggest, total }>}
    */
-  async search (query) {
-    debug('exists(query = %j)', query);
+  async search (searchBody, { from, size } = {}) {
+    debug('exists(searchBody = %j)', searchBody);
 
-    const sanitizedQuery = objectRecursiveMap(query, (key) => {
+    const forbiddenProperties = [
+      'multi_match',
+      ...Object.keys(storageMapping.users.properties)
+    ];
+
+    const sanitizedQuery = objectRecursiveMap(searchBody, key => {
       if (key === 'username') {
         return { key: '_id' };
       }
 
-      const forbiddenProperties = Object.keys(storageMapping.users.properties);
-      
-      if ( forbiddenProperties.find(property => property === key ) ) {
+      if (forbiddenProperties.includes(key)) {
```

Comment:
Again, I would prefer to have a whitelist allowing only `username` than a maybe incomplete blacklist that we will need to keep up to date with new ES features as well

Reply:
I agree that the whitelist is better. To solve this, I could ensure that username is present in the request and throw an error if not or if it's one of the blacklist properties (which is up to date since any new property for this plugin will be blacklisted).
But I wouldn't know how to have only a whitelist which would contain the whole ES search language + username since the user might search on multiple criterias

Reply:
Hmm I see why it's difficult now. I'm really not comfortable with that actually :( 

We should be sure that there is no way to query another param than `username` with `multi_match` or any query like this.

Your proposal seems good to me: ensuring username + blacklist properties. Let's try with that